src/program/utils/debrid_cdn_url.py

```
@@ -39,47 +62,87 @@ def validate(self) -> str | None:
-        except httpx.HTTPStatusError as e:
-            status_code = e.response.status_code
-
-            if status_code in (HTTPStatus.NOT_FOUND, HTTPStatus.GONE):
-                self._refresh()
+        except RefreshedURLIdenticalException:
+            # If the URL hasn't changed after refreshing, it is likely dead.
+            # Raise an exception to indicate the link is unavailable to trigger a re-scrape.
+            raise DebridServiceLinkUnavailable(
+                provider=self.provider,
+                link=self.url or "Unknown URL",
+            )
```

Ruff (0.14.8)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)